### DISCUSSION TITLE
K8s scans from Windows host create invalid temporary filenames

### DISCUSSION THREAD

**Discussion by janikgar** [aquasecurity/trivy#4659] [Bugs]
### Description

When running a Kubernetes scan on a Windows host, scanning fails when trying to create temporary files that contain colons, which are not valid in Windows.

It appears that Trivy is creating temporary files with the literal name of the Kubernetes object (in this case, it appears to be the role `system:controller:bootstrap-signer`. It is common in Kubernetes for Roles, ClusterRoles, RoleBindings, and ClusterRoleBindings to contain colons.

Can these be escaped safely?

### Desired Behavior

Scanning of Kubernetes from a Windows host succeeds

### Actual Behavior

Scanning of Kubernetes from a Windows host fails while creating a temporary file

### Reproduction Steps

```bash
1. From a Windows 10 host, run a scan of a Kubernetes namespace with Roles or ClusterRoles that contain colons. (kube-system usually has some)
```

### Target

Kubernetes

### Scanner

Misconfiguration

### Output Format

Table

### Mode

Standalone

### Debug Output

```bash
PS C:\Users\janik> trivy k8s -n kube-system --report summary --components workload --scanners config -d all
2023-06-16T16:09:09.434-0700    DEBUG   Severities: ["UNKNOWN" "LOW" "MEDIUM" "HIGH" "CRITICAL"]
2023-06-16T16:09:10.932-0700    DEBUG   cache dir:  C:\Users\janik\AppData\Local\trivy
66 / 83 [-------------------------------------------------------------------------------------->______________________] 79.52% 3 p/s
2023-06-16T16:09:33.769-0700    FATAL   k8s scan error:
    github.com/aquasecurity/trivy/pkg/k8s/commands.(*runner).run
        /home/runner/work/trivy/trivy/pkg/k8s/commands/run.go:94
  - scanning misconfigurations error:
    github.com/aquasecurity/trivy/pkg/k8s/scanner.(*Scanner).Scan.func2
        /home/runner/work/trivy/trivy/pkg/k8s/scanner/scanner.go:69
  - scan error:
    github.com/aquasecurity/trivy/pkg/k8s/scanner.(*Scanner).scanMisconfigs
        /home/runner/work/trivy/trivy/pkg/k8s/scanner/scanner.go:123
  - creating tmp file error:
    github.com/aquasecurity/trivy/pkg/k8s/scanner.createTempFile
        /home/runner/work/trivy/trivy/pkg/k8s/scanner/io.go:20
  - open C:\Users\janik\AppData\Local\Temp\kube-system-Role-system:controller:bootstrap-signer-2934213283.yaml: The filename, directory name, or volume label syntax is incorrect.
```

### Operating System

Windows 10

### Version

```bash
PS C:\Users\janik> trivy --version
Version: 0.42.1
Vulnerability DB:
  Version: 2
  UpdatedAt: 2023-06-16 18:06:43.585875751 +0000 UTC
  NextUpdate: 2023-06-17 00:06:43.585875351 +0000 UTC
  DownloadedAt: 2023-06-16 22:37:06.6336789 +0000 UTC
Policy Bundle:
  Digest: sha256:030345bff4845bae1c89450da85681bc25c0c40d2144a0d9db5254674d155298
  DownloadedAt: 2023-06-16 22:37:08.2354202 +0000 UTC
```

### Checklist

- [X] Run `trivy --reset`
- [X] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by knqyf263** [comment]
@chen-keinan Can you please take a look?

**Comment by chen-keinan** [comment]
@janikgar thanks you for reporting this, it definitely can be improved for safety, I'm converting it into an issue #4665
